internal/app/handlers.go

```diff
@@ -460,8 +461,60 @@ func (app *Application) home(w http.ResponseWriter, r *http.Request) {
 	component.Render(r.Context(), w)
 }
 
-func (app *Application) searchResults(w http.ResponseWriter, r *http.Request) {
-	component := components.SearchResults("Search", "Hepatitis")
+func (app *Application) formSearch(w http.ResponseWriter, r *http.Request) {
+	r.ParseForm()
+	searchTerm := r.Form["search"][0]
+	searchType := r.Form["options"][0]
+	search(w, r, app.repository, searchTerm, searchType, app.logger)
+}
+
+func (app *Application) directSearch(w http.ResponseWriter, r *http.Request) {
+	searchType := r.URL.Query().Get("type")
+	searchTerm := r.URL.Query().Get("input")
+	search(w, r, app.repository, searchTerm, searchType, app.logger)
+}
+
+func search(w http.ResponseWriter, r *http.Request, rp *repository.Repository, searchTerm, searchType string, logger *slog.Logger) {
```

Comment:
Is there any reason we couldn't add `app` as a receiver for this function to avoid passing the repository and logger?

Reply:
@nickclyde updated!